src/python/grpcio_tests/tests_gevent/unit/close_channel_test.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2021 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from concurrent import futures
+from src.proto.grpc.testing import empty_pb2, messages_pb2, test_pb2_grpc
+import grpc
+import gevent
+import sys
+from gevent.pool import Group
+from tests_gevent.unit._test_server import start_test_server, UNARY_CALL_WITH_SLEEP_VALUE
+
+_UNARY_CALL_METHOD_WITH_SLEEP = '/grpc.testing.TestService/UnaryCallWithSleep'
+
+
+class CloseChannelTest(unittest.TestCase):
```

Comment:
This is a great test that checks the behavior around `GreenletExit`, which is a special exception. Can we try to inject a normal Python exception, and ensure it will populate to application?

Reply:
I added `test_kill_greenlet_with_generic_exception`; it kills the greenlet with `Exception` instead of using the default `GreenletExit` and checks whether the greenlet returns the corresponding exception.